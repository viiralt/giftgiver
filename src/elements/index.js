import styled from 'styled-components';

export const Wrapper = styled.div``;

export const Heading = styled.h2``;

export const Button = styled.button``;

export const GiftList = styled.div``;
