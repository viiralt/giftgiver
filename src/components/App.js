import React, { Component } from 'react';

import Gift from './Gift';
import { Wrapper, Heading, Button, GiftList } from '../elements';

export default class App extends Component {
  state = {
    gifts: [],
  };

  handleClick = () => {
    const { gifts } = this.state;
    const ids = gifts.map(gift => gift.id);
    const maxId = ids.length > 0 ? Math.max(...ids) : 0;
    gifts.push({ id: maxId + 1 });
    this.setState({ gifts });
  };

  render() {
    const { gifts } = this.state;
    return (
      <Wrapper>
        <Heading>Gift Giver</Heading>
        <GiftList>{gifts.map(gift => <li key={gift.id} />)}</GiftList>
        <Button onClick={this.handleClick}>add gift</Button>
        <Gift />
      </Wrapper>
    );
  }
}
