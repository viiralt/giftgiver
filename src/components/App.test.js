import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
import { Button, GiftList } from '../elements';

const wrapper = shallow(<App />);

it('should render the <App /> correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('initialises `state` with an empty list of gifts', () => {
  // App.state.gifts
  expect(wrapper.state().gifts).toEqual([]);
});

it('adds a new gift to `state` when clicking the `add gift` button', () => {
  // assume `add gift` button has been pressed
  wrapper.find(Button).simulate('click');

  expect(wrapper.state().gifts).toEqual([{ id: 1 }]);
});

it('adds a new gift to the render list when clicking on the `add gift` button', () => {
  wrapper.find(Button).simulate('click');

  expect(wrapper.find(GiftList).children().length).toEqual(2);
});
