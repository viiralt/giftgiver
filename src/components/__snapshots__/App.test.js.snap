// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the <App /> correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.h2>
          Gift Giver
</styled.h2>,
        <styled.div>
          
</styled.div>,
        <styled.button
          onClick={[Function]}
>
          add gift
</styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Gift Giver",
        },
        "ref": null,
        "rendered": "Gift Giver",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "add gift",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "add gift",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h2>
            Gift Giver
</styled.h2>,
          <styled.div>
            
</styled.div>,
          <styled.button
            onClick={[Function]}
>
            add gift
</styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Gift Giver",
          },
          "ref": null,
          "rendered": "Gift Giver",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "add gift",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "add gift",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
